src/components/__tests__/Index.js

```diff
@@ -99,3 +114,80 @@ it('renders correctly (with slot used)', () => {
 
   expect(wrapper.html()).toMatchSnapshot();
 });
+
+it('Allows a change in `index-name`', () => {
+  const wrapper = mount(Index, {
+    propsData: {
+      searchClient: {},
+      indexName: 'bla',
+    },
+  });
+
+  wrapper.setProps({
+    indexName: 'doggie_bowl',
+  });
+
+  expect(instantsearch.__helper.setIndex).toHaveBeenCalledTimes(1);
+  expect(instantsearch.__helper.setIndex).toHaveBeenCalledWith('doggie_bowl');
+  expect(instantsearch.__helper.search).toHaveBeenCalledTimes(1);
```

Comment:
Maybe assert that the results are the one you expected instead of relying on `helper.search` to be called?